roles/deploy_bmh/template/bmh.yml.j2

```diff
@@ -0,0 +1,41 @@
+{%- for node in baremetal_nodes if node.split('-')[0] not in [ 'crc', 'controller', 'ocp' ]  %}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ node }}-bmc-secret
+  namespace: openstack
+type: Opaque
+stringData:
+  username: {{ baremetal_nodes[node]['username'] }}
+  password: {{ baremetal_nodes[node]['password'] }}
+
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: {{ node }}
+  namespace: openstack
+labels:
+  app: openstack
+  workload: {{ node.split('-')[0] }}
+spec:
+  bmc:
```

Comment:
Another thing we will need: `disableCertificateVerification`
There is no warranty about the compute's BMCs to have a non-selfsigned certificate provisioned.